File: BirdBlog-backend/BirdBlog-framework/src/main/java/com/birdy/domain/entity/Article.java
package com.birdy.domain.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.time.LocalDateTime;
import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 文章表
 * @TableName bg_article
 */
@TableName(value ="bg_article")
@Data
@NoArgsConstructor
@AllArgsConstructor
@lombok.EqualsAndHashCode(callSuper = false)
public class Article {
    /**
     * 文章ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 标题
     */
    private String title;

    /**
     * URL别名（便于SEO）
     */
    private String slug;

    /**
     * 摘要
     */
    private String summary;

    /**
     * 正文内容
     */
    private String content;

    /**
     * 分类ID
     */
    private Long categoryId;

    /**
     * 作者(用户ID)
     */
    private Long authorId;

    /**
     * 缩略图URL
     */
    private String thumbnail;

    /**
     * 是否置顶
     */
    private Boolean isTop;

    /**
     * 状态：0已发布，1草稿
     */
    private Integer status;

    /**
     * 允许评论
     */
    private Boolean isComment;

    /**
     * 浏览量
     */
    private Long viewCount;

    /**
     * 点赞量
     */
    private Long likeCount;

    /**
     * 评论数
     */
    private Long commentCount;

    /**
     * 发布时间
     */
    private Date publishedTime;

    /**
     * 置顶时间
     */
    private Date pinnedTime;

    /**
     * 创建者
     */
    @TableField(fill = FieldFill.INSERT)
    private String creator;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新者
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private String updater;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 是否删除
     */
    private Boolean deleted;
}

File: admin-frontend/src/api/article.ts
import http from './http'
import type { PageResult } from '@/types'

/**
 * 文章信息
 */
export interface Article {
  id: number
  title: string
  slug?: string
  summary?: string
  content?: string
  categoryId?: number
  categoryName?: string
  authorId?: number
  authorName?: string
  thumbnail?: string
  isTop?: boolean
  status: number
  statusDesc?: string
  isComment?: boolean
  viewCount?: number
  likeCount?: number
  commentCount?: number
  publishedTime?: string
  pinnedTime?: string
  createTime?: string
  updateTime?: string
  creator?: string
  updater?: string
  deleted?: boolean
  tagIds?: number[]
  newTags?: string[]
  newTagsDetail?: { name: string; remark?: string }[]
  newCategoryName?: string
  newTagRemark?: string
  newCategoryRemark?: string
}

/**
 * 文章查询参数
 */
export interface ArticleQueryParams {
  pageNum?: number
  pageSize?: number
  keyword?: string
  categoryId?: number
  status?: number
  authorId?: number
  isTop?: boolean
}


/**
 * 获取文章分页列表
 *
 * @param params 查询参数，包括分页、搜索条件等
 * @returns 分页结果，包含文章列表和分页信息
 */
export function getArticlePage(params: ArticleQueryParams): Promise<PageResult<Article>> {
  return http.get('/admin/article/list', params)
}

/**
 * 获取文章详情
 *
 * @param id 文章ID
 * @returns 文章详细信息
 */
export function getArticleDetail(id: string | number): Promise<Article> {
  return http.get(`/admin/article/${id}`)
}

/**
 * 创建文章
 *
 * @param data 文章数据
 * @returns 创建结果
 */
export function createArticle(data: Partial<Article>): Promise<void> {
  return http.post('/admin/article', data)
}

/**
 * 更新文章
 *
 * @param id 文章ID
 * @param data 文章数据
 * @returns 更新结果
 */
export function updateArticle(id: string | number, data: Partial<Article>): Promise<void> {
  return http.put(`/admin/article/${id}`, data)
}

/**
 * 删除文章
 *
 * @param ids 文章ID，多个ID用逗号分隔
 * @returns 删除结果
 */
export function deleteArticle(ids: string): Promise<void> {
  return http.delete(`/admin/article/${ids}`)
}

/**
 * 发布文章（将草稿状态改为已发布）
 *
 * @param id 文章ID
 * @returns 发布结果
 */
export function publishArticle(id: string | number): Promise<void> {
  return http.put(`/admin/article/${id}/publish`)
}

/**
 * 将文章设为草稿
 */
export function draftArticle(id: string | number): Promise<void> {
  return http.put(`/admin/article/${id}/draft`)
}

/**
 * 置顶/取消置顶文章
 *
 * @param id 文章ID
 * @param isTop 是否置顶
 * @returns 操作结果
 */
export function toggleArticleTop(id: string | number, isTop: boolean): Promise<void> {
  return http.put(`/admin/article/${id}/top`, { isTop })
}
